### PR TITLE
Add InputDialog and tests for it (fixes #2084)

### LINKED ISSUE #2084: Add implementation for InputDialog
and tests for it.

Dialog could look like this: 
image

### PR DESCRIPTION
**Pull request by jkopriva**
# Pull Request Checklist

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?
* Is copyright with correct year in new files?
* Is there a JavaDoc in every class with description of a class and an author(and issue if suitable)?
* Is there a link to github project issue? (if needed)

## Testing
* Does it work?
* Are the tests green on Jenkins? Is there a link to Jenkins build?

### CODE REVIEW

File: tests/org.eclipse.reddeer.jface.test/src/org/eclipse/reddeer/jface/test/dialogs/InputDialogTest.java [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+					s.dispose();
+				}
+				if (inputDialog != null) {
+					inputDialog.close();
+				}
+			}
+		});
+	}
+
+	private class InputDialogImpl extends InputDialog {
```
Comment (odockal) [inline tests/org.eclipse.reddeer.jface.test/src/org/eclipse/reddeer/jface/test/dialogs/InputDialogTest.java:93]: What is it good for? It does not seem to be used.

**Review by odockal — CHANGES_REQUESTED** [review]

**Comment by odockal** [comment]
@jkopriva And rebase please.

**Comment by odockal** [comment]
@jkopriva Today is M1 day, if you wanna have this PR merged, please look at comments ;)

File: tests/org.eclipse.reddeer.jface.test/src/org/eclipse/reddeer/jface/test/dialogs/InputDialogTest.java (hunk 1 above)
Comment (jkopriva) [reply tests/org.eclipse.reddeer.jface.test/src/org/eclipse/reddeer/jface/test/dialogs/InputDialogTest.java]: deleted

**Comment by odockal** [comment]
@jkopriva Rebase this please.

**Review by odockal — APPROVED** [review]

**Pull request merged**